ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/activityFileAnswer.component.ts

```diff
@@ -0,0 +1,163 @@
+        }
+
+        .file-select {
+            text-align: center;
+        }
+
+        .drop-block {
+            width: 200px;
+            height: 150px;
+            border: 1px dashed grey;
+            border-radius: 10px;
+            background-color: whitesmoke;
+            display: flex;
+            flex-direction: column;
+            justify-content: center;
+        }
+
+        .drop-block.file-over {
+            background-color: lightyellow;
+        }
+
+        .upload-icon {
+            font-size: 48px;
+            width: auto;
+            height: auto;
+        }
+
+        .upload-btn {
+            margin: 20px 0;
+        }
+
+        .file-info {
+            padding: 10px 20px;
+        }
+
+        .file-input {
+            display: none;
+        }
+
+        .submit-btn {
+            margin: 20px 0;
+            float: right;
+        }
+        `
+    ]
+})
+export class ActivityFileAnswer implements OnDestroy {
+    @Input() block: ActivityFileQuestionBlock;
+    @Input() readonly: boolean;
+    @Input() studyGuid: string;
+    @Input() activityGuid: string;
+    @Output() valueChanged: EventEmitter<string | null> = new EventEmitter();
+
+    file: File;
+    fileUploadData: FileUploadResponse;
+    isFileSelected: boolean;
+    private ngUnsubscribe = new Subject();
+
+    constructor(private fileUploadService: FileUploadService,
+                private logger: LoggingService) {
+    }
+
+    onFilesSelected(files: File[]): void {
+        const file: File = files[0];
+
+        if (file) {
+            this.file = file;
+
+            const requestBody: FileUploadBody = {
+                questionStableId: this.block.stableId,
+                fileName: file.name,
+                fileSize: file.size,
+                mimeType: file.type
+            };
+
+            this.fileUploadService.getUploadUrl(this.studyGuid, this.activityGuid, requestBody)
+                .pipe(
+                    catchError(err => {
+                        this.logger.logDebug('ActivityFileAnswer getUploadUrl error:', err);
+                        return EMPTY;
+                    }),
+                    takeUntil(this.ngUnsubscribe)
+                )
+                .subscribe((res: FileUploadResponse) => {
+                    this.isFileSelected = true;
+                    this.fileUploadData = res;
+                });
+        }
+    }
+
+    uploadFile(): void {
+        const formData = new FormData();
+        formData.append('file', this.file);
+
+        this.fileUploadService.uploadFile(this.fileUploadData.uploadUrl, formData, this.file.type)
+            .pipe(
+                catchError(err => {
+                    this.logger.logDebug('ActivityFileAnswer uploadFile error:', err);
+                    return EMPTY;
+                }),
+                takeUntil(this.ngUnsubscribe)
+            )
+            .subscribe((res) => {
+                this.block.answer = this.fileUploadData.uploadGuid;
```

Comment:
I'd recommend not changing the component inputs of the object type. I see that this approach existed already in other answer components, but it's not recommended by the angular way to update data. If we mutate the object in the ActivityFileAnswer component it will be mutated in the parent as well which might be not predictable and well tracked. The output should be the legal way to change the parent data. In order to pass the updated data into the ddp-question-prompt component, you can create the new data structure, copy all block properties and update the answer property when you need it. This way the parent data won't be mutated.

Reply:
The thing is that the (legacy) logic is implemented according to the parent mutation fact.
Let's agree to create a 'tech-debt' ticket as to try to fix the logic when we have time 
(not in scope of 'File upload' ticket), shall we ?

Reply:
I have created [DDP-6110 ticket](https://broadinstitute.atlassian.net/browse/DDP-6110) for that and put it in our backlog.